extensions/resteasy-reactive/rest/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/security/EagerSecurityInterceptorHandler.java

```diff
@@ -65,7 +44,25 @@ public static HandlerChainCustomizer newInstance() {
         public List<ServerRestHandler> handlers(Phase phase, ResourceClass resourceClass,
                 ServerResourceMethod serverResourceMethod) {
             if (phase == Phase.AFTER_MATCH) {
-                return Collections.singletonList(new EagerSecurityInterceptorHandler());
+
+                var desc = ResourceMethodDescription.of(serverResourceMethod);
+                var interceptorStorage = Arc.container().instance(EagerSecurityInterceptorStorage.class).get();
+                var interceptor = interceptorStorage.getInterceptor(desc.invokedMethodDesc());
+                if (interceptor == null && desc.fallbackMethodDesc() != null) {
+                    interceptor = interceptorStorage.getInterceptor(desc.fallbackMethodDesc());
+                }
+
+                if (interceptor == null) {
+                    throw new IllegalStateException(
+                            """
+                                    Security annotation placed on resource method '%s#%s' wasn't detected by Quarkus during the build time.
```

Comment:
Should we say `Handler for the security annotation... is not found` or `CDI interceptor for the security annotation...is not found` ?

Reply:
Before I start, I am absolutely ready to improve the message! Let's just discuss how because suggestions are not ideal.

`CDI interceptor for the security annotation...is not found`

this is not related to CDI interceptors

`Handler for the security annotation... is not found`

well, handler cannot be found, we create it in here. What is not found is security check which IMHO is something 99 % users doesn't work with or don't even know about (maybe more, even though it's public API it's something only Quarkus deals with).

context: if this happens, it means there is inconsistency, I don't know how could this inconsistency happen considering we are using same method to find endpoint impl. in `ServerEndpointIndexer` as we do in `ResteasyReactiveProcessor`. But I wonder to have there this check because if such inconsistency indeed happened, it could mean security leak and it's better to stop this during app startup.

another details: I was bit fighting if I should add this exception because it could bite us, there could be scenario I didn't think of and users would see this exception, but I want to be safe.

That's why I suggest something vague like I did currently, because what is not working is implementation detail.

Reply:
Sorry, wrong file, what is not found here is "eager security interceptor" which is "my invention" and I am sure noone knows about it. It is used for `@Tenant` and `@Basic` etc. so that we can assure it runs before eager security checks

`CDI interceptor for the security annotation...is not found`

now I get your suggestion @sberyozkin , but not a CDI interceptor

Reply:
@sberyozkin to sum this up, I think that message which is there now is ok, but please think about alternatives and suggest if you have one. I don't think we can actually describe to user what went wrong, it's rather about forbidding app to start if illegal state happens. Thank you

Reply:
@michalvavrik Sorry, sure, lets leave it as is, hopefully it will never be reported anyway :-)

Reply:
no worry, I don't think you were wrong. thanks for comments, sure, I also hope it never gets reported